packages/components/src/AppSwitcher/AppSwitcher.component.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import I18N_DOMAIN_COMPONENTS from '../constants';
+
+import Action from '../Actions/Action';
+import ActionDropdown from '../Actions/ActionDropdown';
+
+import AppSwitcherCSSModule from './AppSwitcher.scss';
+import { getTheme } from '../theme';
+
+const theme = getTheme(AppSwitcherCSSModule);
+
+export default function AppSwitcher({ label, isSeparated, ...props }) {
+	const { t } = useTranslation(I18N_DOMAIN_COMPONENTS);
+	const className = theme('tc-app-switcher-action', {
+		separated: isSeparated,
+	});
+
+	let ActionComponent;
+	let clickAction;
+	let ariaLabel;
+	if (props && props.items && props.items.length) {
+		ActionComponent = ActionDropdown;
+		ariaLabel = t('APP_SWITCHER', {
+			defaultValue: 'Switch to another application. Current application: {{appName}}',
+			appName: label,
+		});
+	} else {
+		ActionComponent = Action;
+		clickAction = props.onClick;
+	}
```

Comment:
Why do you have this `simple button` case in an AppSwitcher component ?
This should be managed by HeaderBar IMO, because without items it's not an app switcher
```
if(items) 
    then render AppSwitcher
else 
    render a simple button
```

Reply:
I prefer add this behavior in the AppSwitcher container. The headerbar should just show the component and don't add business behavior inside.